piv/key.go

```diff
@@ -210,6 +216,16 @@ func parseAttestation(slotCert *x509.Certificate) (*Attestation, error) {
 			return nil, fmt.Errorf("parsing extension: %v", err)
 		}
 	}
+
+	// The common name of the subject on the certificate ends with the
+	// two-letter slot name.
+	// https://developers.yubico.com/PIV/Introduction/PIV_attestation.html
+	subjectCN := slotCert.Subject.CommonName
+	if len(subjectCN) < 2 {
+		return nil, fmt.Errorf("invalid common name on attestation: %s", subjectCN)
```

Comment:
nit: please don't return an error. This package kinda supports non-YubiKeys which may use a different format.

Reply:
I wasn't aware of this goal given the naming throughout the file and the `A Go YubiKey PIV implementation` title at the top of the readme. It's out of the scope for this PR, but it might be beneficial to clear that up.

Reply:
Sorry, sent #90 to clarify that.

In hindsight, all of the attestation support is pretty tailored to YubiKeys (we implicitly use the YubiKey CA to verify certificates). Feel free to keep the error message.